errors.go

```diff
@@ -87,19 +101,26 @@ type cause struct {
 	msg   string
 }
 
-func (c cause) Error() string { return fmt.Sprintf("%v", c) }
+func (c cause) Error() string { return fmt.Sprintf("%s: %v", c.msg, c.Cause()) }
 func (c cause) Cause() error  { return c.cause }
 
-func (c cause) Format(s fmt.State, verb rune) {
+// wrapper is an error implementation returned by Wrap and Wrapf
+// that implements its own fmt.Formatter.
+type wrapper struct {
+	cause
+	*stack
+}
+
+func (w wrapper) Format(s fmt.State, verb rune) {
 	switch verb {
 	case 'v':
 		if s.Flag('+') {
-			io.WriteString(s, c.msg)
+			fmt.Fprintf(s, "%+v: %s", w.Stacktrace()[0], w.cause.msg)
 			return
 		}
 		fallthrough
```

Comment:
This logic seems strangely different than the code for the `+` flag in `_error.Format`. It seems to me that it should look like this:

```go
		if s.Flag('+') {
			fmt.Fprintf(s, "%+v: ", w.Stacktrace()[0])
		}
		fallthrough
```

The tests cover this code, but only indirectly via testing `errors.Fprint`. A test that calls `fmt.Printf("%+v", wrappedErr)` would help clarify the proper behavior here.